lib/openscap_parser/groups.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require 'openscap_parser/group'
+
+module OpenscapParser
+  # Methods related to finding and saving rule references
+  module Groups
+    def self.included(base)
+      base.class_eval do
+        def groups
+          @groups ||= group_nodes.map do |group_node|
+            OpenscapParser::Group.new(parsed_xml: group_node)
+          end
+        end
+
+        def group_nodes(xpath = ".//Group")
```

Comment:
According to [this](https://www.w3schools.com/xml/xpath_syntax.asp), the `.//` selects all Groups while we need to maintain the nested (tree) structure:
![Screenshot from 2022-01-25 06-56-25](https://user-images.githubusercontent.com/649130/150919560-65875b41-cb9b-4c64-8dde-01412c1a0829.png)

You can have `Groups` under `Groups` and `Rules` can show up on any level as well. Right now we just parse the `Rules` using `.//` into a flat array and the goal is to have the tree-like structure. In order to keep the compatibility with the current implementation on the other side, we should keep the flat version as well. I guess it's doable by choosing the right names for methods :wink: 

I encourage you to look into the [specification](https://csrc.nist.gov/CSRC/media/Publications/nistir/7275/rev-4/final/documents/nistir-7275r4_updated-march-2012_clean.pdf) of XCCDF, I know it's long, but it has everything we need (somewhere).

Reply:
Oh good point, I missed that! I'll fix this.
I've been reading through the specification of XCCDF and it's helped me figure out the Group structure a lot. I was also looking through the couple test xml files in the openscap_parser repo and it's helped me see different examples of Group being used and what the structure looks like.